File: quiz_app_flutter/lib/presentation/quiz_set/intent/quiz_set_intent.dart
// intents
abstract class QuizSetIntent {}

class LoadQuizSet extends QuizSetIntent {
  final String? quizTopic;
  LoadQuizSet(this.quizTopic);
}

class NavigateToQuiz extends QuizSetIntent {
  /*final QuizSetSectionItem data;
  NavigateToQuiz(this.data);*/
}

// navigation events
abstract class QuizSetNavEvent {}

class QuizSetNavEventNavigateToQuiz extends QuizSetNavEvent {
  /*final QuizScreenData data;
  QuizSetNavEventNavigateToQuiz(this.data);*/
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/presentation/quizSets/intent/QuizSetIntent.kt
package com.tnm.quizappkmp.presentation.quizSets.intent

import com.tnm.quizappkmp.domain.model.quizSets.QuizSetData
import com.tnm.quizappkmp.presentation.quiz.route.QuizScreenData

sealed class QuizSetIntent {
    data class LoadQuizSet(val quizTopic: String?) : QuizSetIntent()
    data class NavigateToQuiz(val data: QuizSetData.SectionItem) : QuizSetIntent()
}

sealed class QuizSetNavEvent {
    data class NavigateToQuiz(val data: QuizScreenData) : QuizSetNavEvent()
}